### ISSUE TITLE
Performance on trec dataset drops significantly after using eda

### ISSUE THREAD

**Issue by RenShuhuai-Andy** [jasonwei20/eda_nlp#23]
**model**: BertForSequenceClassification
**train_set size**: 120 of 5452, using `sklearn.model_selection.StratifiedShuffleSplit`  to keep classes corresponds to the original distribution of train set:
```
def split(self, examples, test_size, train_size, n_splits=2, split_idx=0):
    label_map = {"ENTY": 0, "DESC": 1, 'LOC': 2, 'ABBR': 3, 'NUM': 4, 'HUM': 5}
    labels = [label_map[e.label] for e in examples]
    kf = StratifiedShuffleSplit(n_splits=n_splits, test_size=test_size, train_size=train_size, random_state=C.get()['seed'])
    kf = kf.split(list(range(len(examples))), labels)
    for _ in range(split_idx + 1):  # split_idx equal to cv_fold. this loop is used to get i-th fold
        train_idx, valid_idx = next(kf)
    train_dev_set = np.array(examples)
    return list(train_dev_set[train_idx]), list(train_dev_set[valid_idx])
```
**valid_set size**: 180
**test_set size**: all(500)
**alpha**: 0.1

**augment code**:
```
labels = labels.repeat(n_aug)
aug_texts = [eda(text, alpha_sr=alpha, alpha_ri=alpha, alpha_rs=alpha, p_rd=alpha, num_aug=n_aug - 1) for text in texts]
assert len(labels)==len(aug_texts)
```
I have checked that the labels and augmented texts are matched correctly.

**result when `n_aug=16`** (the average of three experiments) :
- before augment: 0.7975
- after augment: 0.7657

**result when `n_aug=8`** (the average of three experiments) :
- before augment: 0.813
- after augment: 0.7044

It's very confusing. In my experiment, it seems that eda performs well when texts are long (such as `imdb` dataset), but has poor performce in datasets like `trec` and `sst5`. Did I make any mistakes in the experiments setting?

**Comment by jasonwei20** [comment]
If you have manually checked several examples of augmented data and their labels and those are correct, then your code for generating the augmented data is probably correct.

I am not sure about how EDA affects the performance of BERT. One reason I could think of that might hurt the performance is if BERT already has the ability to do well on TREC and SST (which it probably does), i.e., there is little or no overfitting on BERT, then introducing some augmentations might actually hurt its performance. To be conservative, you can try using a really low alpha parameter (0.05 or 0.03).

Sorry I can't be more helpful on this occasion. If you find that EDA does not do well versus long or short texts, and want to report some results on that, I'm happy to modify the readme so that others know.

**Comment by RenShuhuai-Andy** [comment]
Thanks very much for your reply. I will do more experiments based on BERT and give feedback if there are new discoveries.
